### PR TITLE
Feature/df 2213  delete data on unregistering user

### PR DESCRIPTION
**Pull request by emina-str**
Delete CandidateProfiles on UserUnregisteredEvent.

OS: https://github.com/alv-ch/alv-onlineform-service/pull/112
JOBROOM: https://github.com/alv-ch/jobroom2/pull/907
CANDIDATE: https://github.com/alv-ch/candidate-service/pull/170
JOBAD: https://github.com/alv-ch/job-ad-service/pull/442

### CODE REVIEW

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/messagebroker/users/UserEventDto.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package ch.admin.seco.service.candidate.service.messagebroker.users;
+
+public class UserEventDto {
+
+    private String userInfoId;
+
+    private String userInfoRole;
```
Comment (cjpsycos) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/service/messagebroker/users/UserEventDto.java]: can be more than one role

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/LoginAsTechnicalUser.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package ch.admin.seco.service.candidate.service;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.METHOD})
+@Retention(RetentionPolicy.RUNTIME)
+public @interface LoginAsTechnicalUser {
```
Comment (cjpsycos) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/service/LoginAsTechnicalUser.java]: where is the imlementation of it?

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/searchprofile/SearchProfileApplicationService.java [hunk 3]
```diff
@@ -134,6 +135,16 @@ public void deleteSearchProfile(SearchProfileId id) {
         return new PageImpl<>(result, pageable, searchProfiles.getTotalElements());
     }
 
+    @IsSysAdmin
+    public void handleUserUnregisteredEvent(String ownerUserId) {
```
Comment (birom) [inline candidate-app/src/main/java/ch/admin/seco/service/candidate/service/searchprofile/SearchProfileApplicationService.java]: I think the name should reflect what this actually do. The ``UserEventReceiverGateway`` handles the event and this method does the "business" logic. eg.: ``deleteSearchProfiles``

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/LoginAsTechnicalUser.java (hunk 2 above)
Comment (emina-str) [reply candidate-app/src/main/java/ch/admin/seco/service/candidate/service/LoginAsTechnicalUser.java]: added :) thanks.

File: candidate-app/src/main/java/ch/admin/seco/service/candidate/service/messagebroker/users/UserEventDto.java (hunk 1 above)
Comment (emina-str) [reply candidate-app/src/main/java/ch/admin/seco/service/candidate/service/messagebroker/users/UserEventDto.java]: it's a copy of Jobroom event, we only pass one single role, either pav, company or jobseeker.

**Pull request merged**